src/network/client.rs

```diff
@@ -130,6 +172,83 @@ impl NetworkClient {
         }
     }
 
+    async fn run_assignments_loop(&self, cancellation_token: CancellationToken) {
+        let mut timer =
+            tokio::time::interval_at(tokio::time::Instant::now(), self.assignment_update_interval);
+
+        timer.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        IntervalStream::new(timer)
+            .take_until(cancellation_token.cancelled_owned())
+            .for_each(|_| async move {
+                let latest_assignment = self
+                    .assignments
+                    .lock()
+                    .last_key_value()
+                    .map(|(assignament_id, _)| assignament_id.clone());
+                let assignment = match Assignment::try_download(
```

Comment:
This is blocking for a very long time (>20 seconds in debug build) because of JSON deserialization. We should run it inside `tokio::spawn_blocking` to not block the entire runtime

Reply:
done